[mds-policy-author][mds-db] Adding bulk read policy and geography metadata endpoints

janedotx opened the pull request:
## PR Checklist

 - [ ] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [ ] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [ ] Provider
- [ ] Agency
- [ ] Audit
- [ ] Policy
- [ ] Compliance
- [ ] Daily
- [ ] Native
- [x] Policy Author

avatarneil commented on .vscode/launch.json:
```diff
@@ -278,6 +278,7 @@
       "request": "launch",
       "name": "Tests: mds-db",
       "program": "${workspaceFolder}/node_modules/mocha/bin/_mocha",
+      "runtimeArgs": ["--harmony"],
```
I think you added this to try and get `async/await` working nicely in the debugger, but IIRC it didn't work? Might as well remove

janedotx replied:
will do

janedotx commented on packages/mds-db/index.ts:
```diff
@@ -1068,20 +1069,39 @@ async function readGeographies(params?: { geography_id?: UUID; get_read_only?: b
 }
 
+async function readBulkGeographyMetadata(params?: {
+  geography_id?: UUID
```
per our in-person discussion yesterday, i will remove this and add a `readSingleGeography` function to `mds-db`

janedotx commented on packages/mds-db/index.ts:
```diff
@@ -1213,6 +1240,44 @@ async function readPolicies(params?: {
 }
 
+async function readBulkPolicyMetadata(params?: {
+  policy_id?: UUID
```
They're used in the call to `readPolicies` though.

janedotx commented on packages/mds-db/index.ts:
```diff
@@ -1134,32 +1154,39 @@ async function deleteGeography(geography_id: UUID) {
 }
 
-async function readGeographyMetadata(geography_id: UUID): Promise<GeographyMetadata> {
+async function readSingleGeographyMetadata(geography_id: UUID): Promise<GeographyMetadata> {
```
Well, I don't think it makes a lot of sense to have additional params if you know you're expecting a single `Geography` that matches the UUID that you sent. For bulk geography reads, I think it makes sense to be able to sort the published/read-only ones from the unpublished ones.

Now that you've pointed it out, it is true we don't have pagination params for bulk reads. Do you think I should add those?

janedotx commented on packages/mds-db/index.ts:
```diff
@@ -1068,20 +1069,39 @@ async function readGeographies(params?: { geography_id?: UUID; get_read_only?: b
+
+    return rows.map(row => {
+      /* eslint-disable-next-line no-param-reassign */
+      delete row.id
```
Acknowledged.

janedotx commented:
@avatarneil addressed feedback. Also the PUTs verbs do creates AND updates now for the policy and geography metadata. Not having those earlier was a derp on my part

avatarneil commented:
@janedotx Looks like the linter is detecting problems on CI/CD

avatarneil commented:
@twelch would you mind reviewing this and verifying (at least by checking out the test cases) that it behaves how you expect before we merge?

janedotx commented:
> @twelch would you mind reviewing this and verifying (at least by checking out the test cases) that it behaves how you expect before we merge?

Tim and Owen have told me they'd prefer to hit it on sandbox.
